### ISSUE TITLE
iOS Crash: mbgl::LineBucket::drawLines(mbgl::LineShader&, mbgl::gl::ObjectStore&)

### ISSUE THREAD

**Issue by devinross** [mapbox/mapbox-gl-native#7947]
**Platform:** iOS
**Mapbox SDK version:** mapbox-ios-sdk-3.3.5

### Steps to trigger behavior

Saw a crash report from a customer:

```
Incident Identifier: 4624A293-AE4E-43D9-9FC0-E3B914E96793
CrashReporter Key:   253B6D60-EFC8-4A69-83C1-018A4C3C9089
Hardware Model:      iPhone7,1
Process:         Sweetgreen [3451]
Path:            /var/containers/Bundle/Application/42C5F6AA-E931-416A-93E1-604F820EE0CA/Sweetgreen.app/Sweetgreen
Identifier:      com.sweetgreen.sweetgreen
Version:         2.3.3 (2.3.3.1077)
Code Type:       ARM-64
Parent Process:  ??? [1]

Date/Time:       2017-01-19T17:28:49Z
Launch Time:     2017-01-19T17:19:14Z
OS Version:      iPhone OS 9.3.5 (13G36)
Report Version:  104

Exception Type:  SIGSEGV
Exception Codes: SEGV_ACCERR at 0x0
Crashed Thread:  0

Thread 0 Crashed:
0   libsystem_platform.dylib             0x00000001811b62a0 _platform_memmove + 176
1   AGXGLDriver                          0x0000000193103f6c glrAGXRenderVertexArray(GLDContextRec*, unsigned int, unsigned int, int, int, unsigned int, void const*, int, void const*) + 1196
2   GLEngine                             0x0000000193ea6ea4 glDrawElements_ACC_ES2Exec + 424
3   Mapbox                               0x0000000100df8364 mbgl::LineBucket::drawLines(mbgl::LineShader&, mbgl::gl::ObjectStore&) (line_bucket.cpp:472)
4   Mapbox                               0x0000000100dfe604 mbgl::Painter::renderLine(mbgl::LineBucket&, mbgl::style::LineLayer const&, mbgl::UnwrappedTileID const&, std::__1::array<double, 16ul> const&) (painter_line.cpp:163)
5   Mapbox                               0x0000000100dfa694 void mbgl::Painter::renderPass<std::__1::__wrap_iter<mbgl::RenderItem const*> >(mbgl::RenderPass, std::__1::__wrap_iter<mbgl::RenderItem const*>, std::__1::__wrap_iter<mbgl::RenderItem const*>, int, signed char) (painter.cpp:272)
6   Mapbox                               0x0000000100df9a34 mbgl::Painter::render(mbgl::style::Style const&, mbgl::FrameData const&, mbgl::SpriteAtlas&) (painter.cpp:193)
7   Mapbox                               0x0000000100de79e8 mbgl::Map::Impl::render() (map.cpp:263)
8   Mapbox                               0x0000000100de7858 mbgl::Map::render() (map.cpp:175)
9   Mapbox                               0x0000000100d6b7f4 -[MGLMapView glkView:drawInRect:] (MGLMapView.mm:857)
10  GLKit                                0x0000000189bf536c -[GLKView _display:] + 328
11  Mapbox                               0x0000000100d6bf90 -[MGLMapView updateFromDisplayLink] (MGLMapView.mm:984)
12  QuartzCore                           0x000000018409422c CA::Display::DisplayLinkItem::dispatch() + 36
13  QuartzCore                           0x00000001840940e0 CA::Display::DisplayLink::dispatch_items(unsigned long long, unsigned long long, unsigned long long) + 420
14  IOKit                                0x00000001817d1e54 IODispatchCalloutFromCFMessage + 368
15  CoreFoundation                       0x00000001814f9030 __CFMachPortPerform + 176
16  CoreFoundation                       0x00000001815117d4 __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE1_PERFORM_FUNCTION__ + 52
17  CoreFoundation                       0x0000000181510f0c __CFRunLoopDoSource1 + 432
18  CoreFoundation                       0x000000018150ec64 __CFRunLoopRun + 1796
19  CoreFoundation                       0x0000000181438c50 CFRunLoopRunSpecific + 380
20  GraphicsServices                     0x0000000182d20088 GSEventRunModal + 176
21  UIKit                                0x0000000186726088 UIApplicationMain + 200
22  Sweetgreen                           0x0000000100243c00 main (AppDelegate.swift:22)
23  ???                                  0x0000000180fd68b8 0x0 + 0

Thread 1:
0   libsystem_kernel.dylib               0x00000001810f54d8 kevent_qos + 8
1   libdispatch.dylib                    0x0000000180fa7648 _dispatch_mgr_thread + 48

Thread 2:
0   ???                                  0x00000001200a6dcc 0x0 + 0
1   ???                                  0x0000000120098220 0x0 + 0
2   ???                                  0x00000001200a2d1c 0x0 + 0
3   ???                                  0x00000001200a16b0 0x0 + 0
4   ???                                  0x0000000120098a0c 0x0 + 0
5   ???                                  0x000000012009df78 0x0 + 0
6   ???                                  0x0000000180fd56a0 0x0 + 0
7   CFNetwork                            0x0000000181ccb6ac symptom_framework_init + 80
8   CFNetwork                            0x0000000181ccb774 ___ZL16getSymptomHandlev_block_invoke + 20
9   libdispatch.dylib                    0x0000000180fa547c _dispatch_client_callout + 12
10  libdispatch.dylib                    0x0000000180fa62c0 dispatch_once_f + 76
11  CFNetwork                            0x0000000181ccb1c4 __CFNReportRedirectSymptom + 464
12  CFNetwork                            0x0000000181bfed48 HTTPProtocol::attemptRedirection(__CFHTTPMessage*) + 1320
13  CFNetwork                            0x0000000181bfa604 HTTPProtocol::performHeaderRead(__CFHTTPMessage*) + 1120
14  CFNetwork                            0x0000000181bfa10c HTTPProtocol::handleStreamEvent(__CFHTTPMessage*, dispatch_data_s*, CFStreamError const*) + 496
15  CFNetwork                            0x0000000181bd01d0 HTTPNetStreamInfo::_readStreamClientCallBack(__CFReadStream*, unsigned long) + 308
16  CFNetwork                            0x0000000181ca6800 CFNetworkReadStream::_readStreamClientCallBackCallBack(__CFReadStream*, unsigned long, void*) + 76
17  CoreFoundation                       0x00000001814a9bc0 _signalEventSync + 188
18  CoreFoundation                       0x00000001814a9adc _cfstream_shared_signalEventSync + 440
19  CoreFoundation                       0x000000018151109c __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ + 20
20  CoreFoundation                       0x0000000181510ab0 __CFRunLoopDoSources0 + 408
21  CoreFoundation                       0x000000018150e830 __CFRunLoopRun + 720
22  CoreFoundation                       0x0000000181438c50 CFRunLoopRunSpecific + 380
23  CFNetwork                            0x0000000181bb9bcc +[NSURLConnection(Loader) _resourceLoadLoop:] + 408
24  Foundation                           0x0000000181f2fe4c __NSThread__start__ + 996
25  libsystem_pthread.dylib              0x00000001811bfb28 _pthread_body + 152
26  libsystem_pthread.dylib              0x00000001811bfa8c _pthread_start + 152
27  libsystem_pthread.dylib              0x00000001811bd028 thread_start + 0

Thread 3:
0   libsystem_kernel.dylib               0x00000001810d8fd8 mach_msg_trap + 8
1   CoreFoundation                       0x0000000181510c60 __CFRunLoopServiceMachPort + 192
2   CoreFoundation                       0x000000018150e964 __CFRunLoopRun + 1028
3   CoreFoundation                       0x0000000181438c50 CFRunLoopRunSpecific + 380
4   Foundation                           0x0000000181e48cfc -[NSRunLoop(NSRunLoop) runMode:beforeDate:] + 304
5   Foundation                           0x0000000181e9e030 -[NSRunLoop(NSRunLoop) run] + 84
6   Sweetgreen                           0x0000000100391738 +[GAI threadMain:] + 68
7   Foundation                           0x0000000181f2fe4c __NSThread__start__ + 996
8   libsystem_pthread.dylib              0x00000001811bfb28 _pthread_body + 152
9   libsystem_pthread.dylib              0x00000001811bfa8c _pthread_start + 152
10  libsystem_pthread.dylib              0x00000001811bd028 thread_start + 0

Thread 4:
0   libsystem_kernel.dylib               0x00000001810d8fd8 mach_msg_trap + 8
1   CoreFoundation                       0x0000000181510c60 __CFRunLoopServiceMachPort + 192
2   CoreFoundation                       0x000000018150e964 __CFRunLoopRun + 1028
3   CoreFoundation                       0x0000000181438c50 CFRunLoopRunSpecific + 380
4   CoreFoundation                       0x00000001814863a4 CFRunLoopRun + 108
5   CoreMotion                           0x000000018740b41c 0x1873b8000 + 341020
6   libsystem_pthread.dylib              0x00000001811bfb28 _pthread_body + 152
7   libsystem_pthread.dylib              0x00000001811bfa8c _pthread_start + 152
8   libsystem_pthread.dylib              0x00000001811bd028 thread_start + 0

Thread 5:
0   libsystem_kernel.dylib               0x00000001810f4344 __select + 8
1   libsystem_pthread.dylib              0x00000001811bfb28 _pthread_body + 152
2   libsystem_pthread.dylib              0x00000001811bfa8c _pthread_start + 152
3   libsystem_pthread.dylib              0x00000001811bd028 thread_start + 0

Thread 6:
0   libsystem_kernel.dylib               0x00000001810d8fd8 mach_msg_trap + 8
1   CoreFoundation                       0x0000000181510c60 __CFRunLoopServiceMachPort + 192
2   CoreFoundation                       0x000000018150e964 __CFRunLoopRun + 1028
3   CoreFoundation                       0x0000000181438c50 CFRunLoopRunSpecific + 380
4   WebCore                              0x000000018542a108 RunWebThread(void*) + 452
5   libsystem_pthread.dylib              0x00000001811bfb28 _pthread_body + 152
6   libsystem_pthread.dylib              0x00000001811bfa8c _pthread_start + 152
7   libsystem_pthread.dylib              0x00000001811bd028 thread_start + 0

Thread 7:
0   libsystem_kernel.dylib               0x00000001810f3f24 __psynch_cvwait + 8
1   libc++.1.dylib                       0x0000000180b4b42c std::__1::condition_variable::wait(std::__1::unique_lock<std::__1::mutex>&) + 52
2   JavaScriptCore                       0x0000000184f10d5c JSC::GCThread::waitForNextPhase() + 140
3   JavaScriptCore                       0x0000000184f10df4 JSC::GCThread::gcThreadMain() + 80
4   JavaScriptCore                       0x0000000184be6614 WTF::threadEntryPoint(void*) + 208
5   JavaScriptCore                       0x0000000184be6524 WTF::wtfThreadEntryPoint(void*) + 20
6   libsystem_pthread.dylib              0x00000001811bfb28 _pthread_body + 152
7   libsystem_pthread.dylib              0x00000001811bfa8c _pthread_start + 152
8   libsystem_pthread.dylib              0x00000001811bd028 thread_start + 0

Thread 8:
0   libsystem_kernel.dylib               0x00000001810d8fd8 mach_msg_trap + 8
1   CoreFoundation                       0x0000000181510c60 __CFRunLoopServiceMachPort + 192
2   CoreFoundation                       0x000000018150e964 __CFRunLoopRun + 1028
3   CoreFoundation                       0x0000000181438c50 CFRunLoopRunSpecific + 380
4   CoreFoundation                       0x00000001814863a4 CFRunLoopRun + 108
5   Mapbox                               0x0000000100e9ac64 void mbgl::util::Thread<mbgl::DefaultFileSource::Impl>::run<std::__1::tuple<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, unsigned long long&>, 0ul, 1ul>(std::__1::tuple<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, unsigned long long&>&&, std::__1::integer_sequence<unsigned long, 0ul, 1ul>) (thread.hpp:126)
6   Mapbox                               0x0000000100e9ab84 std::__1::__thread_proxy<std::__1::tuple<mbgl::util::Thread<mbgl::DefaultFileSource::Impl>::Thread<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, unsigned long long&>(mbgl::util::ThreadContext const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&&&, unsigned long long&&&)::'lambda'()> >(void*, void*) (thread.hpp:110)
7   libsystem_pthread.dylib              0x00000001811bfb28 _pthread_body + 152
8   libsystem_pthread.dylib              0x00000001811bfa8c _pthread_start + 152
9   libsystem_pthread.dylib              0x00000001811bd028 thread_start + 0

Thread 9:
0   libsystem_kernel.dylib               0x00000001810d8fd8 mach_msg_trap + 8
1   CoreFoundation                       0x0000000181510c60 __CFRunLoopServiceMachPort + 192
2   CoreFoundation                       0x000000018150e964 __CFRunLoopRun + 1028
3   CoreFoundation                       0x0000000181438c50 CFRunLoopRunSpecific + 380
4   CoreFoundation                       0x00000001814863a4 CFRunLoopRun + 108
5   Mapbox                               0x0000000100e9597c void mbgl::util::Thread<mbgl::AssetFileSource::Impl>::run<std::__1::tuple<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&>, 0ul>(std::__1::tuple<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&>&&, std::__1::integer_sequence<unsigned long, 0ul>) (thread.hpp:126)
6   Mapbox                               0x0000000100e958d0 std::__1::__thread_proxy<std::__1::tuple<mbgl::util::Thread<mbgl::AssetFileSource::Impl>::Thread<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&>(mbgl::util::ThreadContext const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&&&)::'lambda'()> >(void*, void*) (thread.hpp:110)
7   libsystem_pthread.dylib              0x00000001811bfb28 _pthread_body + 152
8   libsystem_pthread.dylib              0x00000001811bfa8c _pthread_start + 152
9   libsystem_pthread.dylib              0x00000001811bd028 thread_start + 0

Thread 10:
0   libsystem_kernel.dylib               0x00000001810d8fd8 mach_msg_trap + 8
1   CoreFoundation                       0x0000000181510c60 __CFRunLoopServiceMachPort + 192
2   CoreFoundation                       0x000000018150e964 __CFRunLoopRun + 1028
3   CoreFoundation                       0x0000000181438c50 CFRunLoopRunSpecific + 380
4   CoreFoundation                       0x00000001814863a4 CFRunLoopRun + 108
5   Mapbox                               0x0000000100e8c154 void mbgl::util::Thread<mbgl::Worker::Impl>::run<std::__1::tuple<> >(std::__1::tuple<>&&, std::__1::integer_sequence<unsigned long>) (thread.hpp:126)
6   Mapbox                               0x0000000100e8c0b8 std::__1::__thread_proxy<std::__1::tuple<mbgl::util::Thread<mbgl::Worker::Impl>::Thread<>(mbgl::util::ThreadContext const&)::'lambda'()> >(void*, void*) (thread.hpp:110)
7   libsystem_pthread.dylib              0x00000001811bfb28 _pthread_body + 152
8   libsystem_pthread.dylib              0x00000001811bfa8c _pthread_start + 152
9   libsystem_pthread.dylib              0x00000001811bd028 thread_start + 0

Thread 11:
0   libsystem_kernel.dylib               0x00000001810d8fd8 mach_msg_trap + 8
1   CoreFoundation                       0x0000000181510c60 __CFRunLoopServiceMachPort + 192
2   CoreFoundation                       0x000000018150e964 __CFRunLoopRun + 1028
3   CoreFoundation                       0x0000000181438c50 CFRunLoopRunSpecific + 380
4   CoreFoundation                       0x00000001814863a4 CFRunLoopRun + 108
5   Mapbox                               0x0000000100e8c154 void mbgl::util::Thread<mbgl::Worker::Impl>::run<std::__1::tuple<> >(std::__1::tuple<>&&, std::__1::integer_sequence<unsigned long>) (thread.hpp:126)
6   Mapbox                               0x0000000100e8c0b8 std::__1::__thread_proxy<std::__1::tuple<mbgl::util::Thread<mbgl::Worker::Impl>::Thread<>(mbgl::util::ThreadContext const&)::'lambda'()> >(void*, void*) (thread.hpp:110)
7   libsystem_pthread.dylib              0x00000001811bfb28 _pthread_body + 152
8   libsystem_pthread.dylib              0x00000001811bfa8c _pthread_start + 152
9   libsystem_pthread.dylib              0x00000001811bd028 thread_start + 0

Thread 12:
0   libsystem_kernel.dylib               0x00000001810d8fd8 mach_msg_trap + 8
1   CoreFoundation                       0x0000000181510c60 __CFRunLoopServiceMachPort + 192
2   CoreFoundation                       0x000000018150e964 __CFRunLoopRun + 1028
3   CoreFoundation                       0x0000000181438c50 CFRunLoopRunSpecific + 380
4   CoreFoundation                       0x00000001814863a4 CFRunLoopRun + 108
5   Mapbox                               0x0000000100e8c154 void mbgl::util::Thread<mbgl::Worker::Impl>::run<std::__1::tuple<> >(std::__1::tuple<>&&, std::__1::integer_sequence<unsigned long>) (thread.hpp:126)
6   Mapbox                               0x0000000100e8c0b8 std::__1::__thread_proxy<std::__1::tuple<mbgl::util::Thread<mbgl::Worker::Impl>::Thread<>(mbgl::util::ThreadContext const&)::'lambda'()> >(void*, void*) (thread.hpp:110)
7   libsystem_pthread.dylib              0x00000001811bfb28 _pthread_body + 152
8   libsystem_pthread.dylib              0x00000001811bfa8c _pthread_start + 152
9   libsystem_pthread.dylib              0x00000001811bd028 thread_start + 0

Thread 13:
0   libsystem_kernel.dylib               0x00000001810d8fd8 mach_msg_trap + 8
1   CoreFoundation                       0x0000000181510c60 __CFRunLoopServiceMachPort + 192
2   CoreFoundation                       0x000000018150e964 __CFRunLoopRun + 1028
3   CoreFoundation                       0x0000000181438c50 CFRunLoopRunSpecific + 380
4   CoreFoundation                       0x00000001814863a4 CFRunLoopRun + 108
5   Mapbox                               0x0000000100e8c154 void mbgl::util::Thread<mbgl::Worker::Impl>::run<std::__1::tuple<> >(std::__1::tuple<>&&, std::__1::integer_sequence<unsigned long>) (thread.hpp:126)
6   Mapbox                               0x0000000100e8c0b8 std::__1::__thread_proxy<std::__1::tuple<mbgl::util::Thread<mbgl::Worker::Impl>::Thread<>(mbgl::util::ThreadContext const&)::'lambda'()> >(void*, void*) (thread.hpp:110)
7   libsystem_pthread.dylib              0x00000001811bfb28 _pthread_body + 152
8   libsystem_pthread.dylib              0x00000001811bfa8c _pthread_start + 152
9   libsystem_pthread.dylib              0x00000001811bd028 thread_start + 0

Thread 14:
0   libsystem_kernel.dylib               0x00000001810f4b48 __workq_kernreturn + 8
1   libsystem_pthread.dylib              0x00000001811bd020 start_wqthread + 0

Thread 15:
0   libsystem_kernel.dylib               0x00000001810f4b48 __workq_kernreturn + 8
1   libsystem_pthread.dylib              0x00000001811bd020 start_wqthread + 0

Thread 16:
0   libsystem_kernel.dylib               0x00000001810f4b48 __workq_kernreturn + 8
1   libsystem_pthread.dylib              0x00000001811bd020 start_wqthread + 0

Thread 17:
0   libsystem_kernel.dylib               0x00000001810f4b48 __workq_kernreturn + 8
1   libsystem_pthread.dylib              0x00000001811bd020 start_wqthread + 0

Thread 18:
0   libsystem_kernel.dylib               0x00000001810f4b48 __workq_kernreturn + 8
1   libsystem_pthread.dylib              0x00000001811bd020 start_wqthread + 0

Thread 19:
0   libsystem_kernel.dylib               0x00000001810f4b48 __workq_kernreturn + 8
1   libsystem_pthread.dylib              0x00000001811bd020 start_wqthread + 0

Thread 20:
0   libsystem_kernel.dylib               0x00000001810f4b48 __workq_kernreturn + 8
1   libsystem_pthread.dylib              0x00000001811bd020 start_wqthread + 0

Thread 21:
0   libsystem_kernel.dylib               0x00000001810d8fd8 mach_msg_trap + 8
1   CoreFoundation                       0x0000000181510c60 __CFRunLoopServiceMachPort + 192
2   CoreFoundation                       0x000000018150e964 __CFRunLoopRun + 1028
3   CoreFoundation                       0x0000000181438c50 CFRunLoopRunSpecific + 380
4   CoreFoundation                       0x00000001814863a4 CFRunLoopRun + 108
5   Mapbox                               0x0000000100e8c154 void mbgl::util::Thread<mbgl::Worker::Impl>::run<std::__1::tuple<> >(std::__1::tuple<>&&, std::__1::integer_sequence<unsigned long>) (thread.hpp:126)
6   Mapbox                               0x0000000100e8c0b8 std::__1::__thread_proxy<std::__1::tuple<mbgl::util::Thread<mbgl::Worker::Impl>::Thread<>(mbgl::util::ThreadContext const&)::'lambda'()> >(void*, void*) (thread.hpp:110)
7   libsystem_pthread.dylib              0x00000001811bfb28 _pthread_body + 152
8   libsystem_pthread.dylib              0x00000001811bfa8c _pthread_start + 152
9   libsystem_pthread.dylib              0x00000001811bd028 thread_start + 0

Thread 22:
0   libsystem_kernel.dylib               0x00000001810d8fd8 mach_msg_trap + 8
1   CoreFoundation                       0x0000000181510c60 __CFRunLoopServiceMachPort + 192
2   CoreFoundation                       0x000000018150e964 __CFRunLoopRun + 1028
3   CoreFoundation                       0x0000000181438c50 CFRunLoopRunSpecific + 380
4   CoreFoundation                       0x00000001814863a4 CFRunLoopRun + 108
5   Mapbox                               0x0000000100e8c154 void mbgl::util::Thread<mbgl::Worker::Impl>::run<std::__1::tuple<> >(std::__1::tuple<>&&, std::__1::integer_sequence<unsigned long>) (thread.hpp:126)
6   Mapbox                               0x0000000100e8c0b8 std::__1::__thread_proxy<std::__1::tuple<mbgl::util::Thread<mbgl::Worker::Impl>::Thread<>(mbgl::util::ThreadContext const&)::'lambda'()> >(void*, void*) (thread.hpp:110)
7   libsystem_pthread.dylib              0x00000001811bfb28 _pthread_body + 152
8   libsystem_pthread.dylib              0x00000001811bfa8c _pthread_start + 152
9   libsystem_pthread.dylib              0x00000001811bd028 thread_start + 0

Thread 23:
0   libsystem_kernel.dylib               0x00000001810d8fd8 mach_msg_trap + 8
1   CoreFoundation                       0x0000000181510c60 __CFRunLoopServiceMachPort + 192
2   CoreFoundation                       0x000000018150e964 __CFRunLoopRun + 1028
3   CoreFoundation                       0x0000000181438c50 CFRunLoopRunSpecific + 380
4   CoreFoundation                       0x00000001814863a4 CFRunLoopRun + 108
5   Mapbox                               0x0000000100e8c154 void mbgl::util::Thread<mbgl::Worker::Impl>::run<std::__1::tuple<> >(std::__1::tuple<>&&, std::__1::integer_sequence<unsigned long>) (thread.hpp:126)
6   Mapbox                               0x0000000100e8c0b8 std::__1::__thread_proxy<std::__1::tuple<mbgl::util::Thread<mbgl::Worker::Impl>::Thread<>(mbgl::util::ThreadContext const&)::'lambda'()> >(void*, void*) (thread.hpp:110)
7   libsystem_pthread.dylib              0x00000001811bfb28 _pthread_body + 152
8   libsystem_pthread.dylib              0x00000001811bfa8c _pthread_start + 152
9   libsystem_pthread.dylib              0x00000001811bd028 thread_start + 0

Thread 24:
0   libsystem_kernel.dylib               0x00000001810d8fd8 mach_msg_trap + 8
1   CoreFoundation                       0x0000000181510c60 __CFRunLoopServiceMachPort + 192
2   CoreFoundation                       0x000000018150e964 __CFRunLoopRun + 1028
3   CoreFoundation                       0x0000000181438c50 CFRunLoopRunSpecific + 380
4   CoreFoundation                       0x00000001814863a4 CFRunLoopRun + 108
5   Mapbox                               0x0000000100e8c154 void mbgl::util::Thread<mbgl::Worker::Impl>::run<std::__1::tuple<> >(std::__1::tuple<>&&, std::__1::integer_sequence<unsigned long>) (thread.hpp:126)
6   Mapbox                               0x0000000100e8c0b8 std::__1::__thread_proxy<std::__1::tuple<mbgl::util::Thread<mbgl::Worker::Impl>::Thread<>(mbgl::util::ThreadContext const&)::'lambda'()> >(void*, void*) (thread.hpp:110)
7   libsystem_pthread.dylib              0x00000001811bfb28 _pthread_body + 152
8   libsystem_pthread.dylib              0x00000001811bfa8c _pthread_start + 152
9   libsystem_pthread.dylib              0x00000001811bd028 thread_start + 0

Thread 25:
0   libsystem_kernel.dylib               0x00000001810f4b48 __workq_kernreturn + 8
1   libsystem_pthread.dylib              0x00000001811bd020 start_wqthread + 0

Thread 26:
0   libsystem_pthread.dylib              0x00000001811bd01c start_wqthread + 0

Thread 27:
0   libsystem_kernel.dylib               0x00000001810f4b48 __workq_kernreturn + 8
1   libsystem_pthread.dylib              0x00000001811bd020 start_wqthread + 0

Thread 28:
0   libsystem_kernel.dylib               0x00000001810f4b48 __workq_kernreturn + 8
1   libsystem_pthread.dylib              0x00000001811bd020 start_wqthread + 0

Thread 29:
0   libsystem_kernel.dylib               0x00000001810f4b48 __workq_kernreturn + 8
1   libsystem_pthread.dylib              0x00000001811bd020 start_wqthread + 0

Thread 30:
0   libsystem_kernel.dylib               0x00000001810f4b48 __workq_kernreturn + 8
1   libsystem_pthread.dylib              0x00000001811bd020 start_wqthread + 0

Thread 31:
0   libsystem_kernel.dylib               0x00000001810f441c __semwait_signal + 8
1   libc++.1.dylib                       0x0000000180b893b4 std::__1::this_thread::sleep_for(std::__1::chrono::duration<long long, std::__1::ratio<1l, 1000000000l> > const&) + 80
2   JavaScriptCore                       0x000000018515e768 bmalloc::Heap::scavenge(std::__1::unique_lock<bmalloc::StaticMutex>&, std::__1::chrono::duration<long long, std::__1::ratio<1l, 1000l> >) + 400
3   JavaScriptCore                       0x000000018515e340 bmalloc::Heap::concurrentScavenge() + 80
4   JavaScriptCore                       0x0000000185160ad8 bmalloc::AsyncTask<bmalloc::Heap, void (bmalloc::Heap::*)()>::entryPoint() + 96
5   JavaScriptCore                       0x0000000185160a68 bmalloc::AsyncTask<bmalloc::Heap, void (bmalloc::Heap::*)()>::pthreadEntryPoint(void*) + 8
6   libsystem_pthread.dylib              0x00000001811bfb28 _pthread_body + 152
7   libsystem_pthread.dylib              0x00000001811bfa8c _pthread_start + 152
8   libsystem_pthread.dylib              0x00000001811bd028 thread_start + 0
```

**Comment by friedbunny** [comment]
Again, rather similar to #5731 (and #7948).

/cc @jfirebaugh

**Comment by jfirebaugh** [comment]
Again, line numbers indicate this is not 3.4.0. @devinross can you verify?

**Comment by devinross** [comment]
I was running: mapbox-ios-sdk-3.3.5

**Comment by jfirebaugh** [comment]
Great, thanks for confirming. This issue was fixed in 3.3.6.
